client/state/preferences/test/reducer.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_FETCH,
+	PREFERENCES_FETCH_SUCCESS,
+	PREFERENCES_FETCH_FAILURE,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+import reducer, { createReducerForPreferenceKey, fetching } from '../reducer';
+import { DEFAULT_PREFERENCES, USER_SETTING_KEY } from '../constants';
+
+describe( 'reducer', () => {
+	it( 'should export expected reducer keys', () => {
+		expect( reducer( undefined, {} ) ).to.have.keys( [
+			'values',
+			'fetching'
+		] );
+	} );
+
+	describe( '#values()', () => {
+		it( 'should export reducer keys defined in constants/DEFAULT_PREFERENCES', () => {
+			expect( reducer( undefined, {} ).values ).to.have.keys( Object.keys( DEFAULT_PREFERENCES ) );
+		} );
+
+		it( 'should use DEFAULT_PREFERENCES from constants as a default', () => {
+			expect( reducer( undefined, {} ).values ).to.deep.equal( DEFAULT_PREFERENCES );
+		} );
+	} );
+
+	describe( '#fetching()', () => {
+		it( 'should default to false', () => {
+			const state = fetching( undefined, false );
+			expect( state ).to.eql( false );
+		} );
+		it( 'should update fetching state on fetch', () => {
+			const state = fetching( undefined, {
+				type: PREFERENCES_FETCH
+			} );
+			expect( state ).to.eql( true );
+		} );
+		it( 'should update fetching state on success', () => {
+			const state = fetching( true, {
+				type: PREFERENCES_FETCH_SUCCESS
+			} );
+			expect( state ).to.eql( false );
+		} );
+		it( 'should update fetching state on failure', () => {
+			const original = { all: true };
```

Comment:
We should always wrap initialState objects with `deepFreeze` to make sure we don't mutate :)